### ISSUE TITLE
Proposal: Add prepare npm hook

### ISSUE THREAD

**Issue by d-asensio** [t-mullen/simple-signal#6]
I think that it's not a good practice to commit distributable sources, since they use to be auto-generated and that could cause lots of git conflicts, and it's an unnecessary waste!

The solution could be:
- Remove the "dist" folder from the repo.
- Add the "dist" folder to the .gitignore file.
- Add a npm hook (prepare) to generate the distributable sources right after publishing the package
- Use [GitHub's releases](https://help.github.com/articles/creating-releases/) to allow users download the distributable files.
- Document the process on the contributing guidelines.

I can do it! So I'll submit a PR if I have time

**Comment by t-mullen** [comment]
You're absolutely right. Github Releases or an npm hook is the correct way to do this.

A PR would be welcome 👍
